gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dform_ttt2.lua

```diff
@@ -122,6 +122,80 @@ local function MakeReset(parent)
 	return reset
 end
 
+---
+-- Adds a slider to the form
+-- @param table data The data for the slider
+-- @note Structure of data = {
+-- label, default, convar, serverConVar, initial, function OnChange(value),
+-- master = { function AddSlave(self, slave) }
+-- }
+-- @return Panel The created textentry
+-- @realm client
+function PANEL:MakeTextEntry(data)
+	local left = vgui.Create("DLabelTTT2", self)
+
+	left:SetText(data.label)
+
+	left.Paint = function(slf, w, h)
+		derma.SkinHook("Paint", "FormLabelTTT2", slf, w, h)
+
+		return true
+	end
+
+	local right = vgui.Create("DTextEntryTTT2", self)
+
+	local reset = MakeReset(self)
+	right:SetResetButton(reset)
+
+	right:SetUpdateOnType(false)
+	right:SetHeightMult(1)
+
+	right.OnGetFocus = function(slf)
+		getHighestParent(self):SetKeyboardInputEnabled(true)
+	end
+
+	right.OnLoseFocus = function(slf)
+		getHighestParent(self):SetKeyboardInputEnabled(false)
+	end
+
+
+
+	right:SetPlaceholderText("")
+	right:SetCurrentPlaceholderText("")
+
+	-- Set default if possible even if the convar could still overwrite it
+	right:SetDefaultValue(data.default)
+	right:SetConVar(data.convar)
+	right:SetServerConVar(data.serverConvar)
+	-- right:SizeToContents()
+	-- right:PerformLayout()
+
+	if not data.convar and not data.serverConvar and data.initial then
+		right:SetValue(data.initial)
+	end
+
+	right.OnValueChanged = function(slf, value)
+		if isfunction(data.OnChange) then
+			-- print("ovc:", slf, value)
+			data.OnChange(slf, value)
+		end
+	end
```

Comment:
Is the convar and value change stuff really needed?

Reply:
for lines 166:175, a lot of those were to do with how dnumslider handled it, so i figured that was convention. i am using this for convar stuff, and i tested without these few lines, it did impact populating the field by default. seems necessary to function.